---
title: 《LEWIN’S GENES XII》目录一览
date: '2018-10-21'
slug: content-of-lewins-genes-x11
categories: 
  - books
tags: 
  - biology
  - genes
  - books
---

## 前言

原文摘录:

Of the diverse ways to study the living world, molecular biology has
been most remarkable in the speed and breadth of its expansion.
New data are acquired daily, and new insights into well-studied
processes come on a scale measured in weeks or months rather
than years. It’s difficult to believe that the first complete organismal
genome sequence was obtained a little over 20 years ago. The
structure and function of genes and genomes and their associated
cellular processes are sometimes elegantly and deceptively simple
but frequently amazingly complex, and no single book can do
justice to the realities and diversities of natural genetic systems.

This book is aimed at advanced students in molecular genetics and
molecular biology. In order to provide the most current
understanding of the rapidly changing subjects in molecular biology,
we have enlisted leading scientists to provide revisions and content
updates in their individual fields of expertise. Their expert
knowledge has been incorporated throughout the text. Much of the
revision and reorganization of this edition follows that of the third
edition of Lewin’s Essential GENES, but there are many updates
and features that are new to this book. This edition follows a logical
flow of topics; in particular, discussion of chromatin organization
and nucleosome structure precedes the discussion of eukaryotic
transcription, because chromosome organization is critical to all
DNA transactions in the cell, and current research in the field of
transcriptional regulation is heavily biased toward the study of the
role of chromatin in this process. Many new figures are included in
this book, some reflecting new developments in the field,
particularly in the topics of chromatin structure and function,
epigenetics, and regulation by noncoding RNA and microRNAs in
eukaryotes.

This book is organized into four parts. **Part I (Genes and
Chromosomes)** comprises Chapters 1 through 8. Chapter 1
serves as an introduction to the structure and function of DNA and
contains basic coverage of DNA replication and gene expression.
Chapter 2 provides information on molecular laboratory
techniques. Chapter 3 introduces the interrupted structures of
eukaryotic genes, and Chapters 4 through 6 discuss genome
structure and evolution. Chapters 7 and 8 discuss the structure of
eukaryotic chromosomes.

**Part II (DNA Replication, Repair, and Recombination)** comprises
Chapters 9 through 16. Chapters 9 through 12 provide detailed
discussions of DNA replication in plasmids, viruses, and prokaryotic
and eukaryotic cells. Chapters 13 through 16 cover recombination
and its roles in DNA repair and the human immune system, with
Chapter 14 discussing DNA repair pathways in detail and Chapter
15 focusing on different types of transposable elements.

**Part III (Transcription and Posttranscriptional Mechanisms)
includes Chapters 17 through 23.** Chapters 17 and 18 provide
more in-depth coverage of bacterial and eukaryotic transcription.
Chapters 19 through 21 are concerned with RNA, discussing
messenger RNA, RNA stability and localization, RNA processing,
and the catalytic roles of RNA. Chapters 22 and 23 discuss
translation and the genetic code.

**Part IV (Gene Regulation)** comprises Chapters 24 through 30. In
Chapter 24, the regulation of bacterial gene expression via
operons is discussed. Chapter 25 covers the regulation of
expression of genes during phage development as they infect
bacterial cells. Chapters 26 through 28 cover eukaryotic gene
regulation, including epigenetic modifications. Finally, Chapters 29
and 30 cover RNA-based control of gene expression in prokaryotes
and eukaryotes.

## 目录

```
Cover Page                                                                      1
Title Page                                                                      2
Copyright Page                                                                  3
Dedication                                                                      6
Brief Contents                                                                  8
Contents                                                                        11
Preface                                                                         36
About the Authors                                                               48
PART I Genes and Chromosomes                                                    58
	Chapter 1 Genes Are DNA and Encode RNAs and Polypeptides                        61
		1.1 Introduction                                                                65
		1.2 DNA Is the Genetic Material of Bacteria and Viruses                         72
		1.3 DNA Is the Genetic Material of Eukaryotic Cells                             77
		1.4 Polynucleotide Chains Have Nitrogenous Bases Linked to a Sugar—Phosphate Backbone    79
		1.5 Supercoiling Affects the Structure of DNA                                   82
		1.6 DNA Is a Double Helix                                                       88
		1.7 DNA Replication Is Semiconservative                                         94
		1.8 Polymerases Act on Separated DNA Strands at the Replication Fork            97
		1.9 Genetic Information Can Be Provided by DNA or RNA                           99
		1.10 Nucleic Acids Hybridize by Base Pairing                                    106
		1.11 Mutations Change the Sequence of DNA                                       111
		1.12 Mutations Can Affect Single Base Pairs or Longer Sequences                 113
		1.13 The Effects of Mutations Can Be Reversed                                   118
		1.14 Mutations Are Concentrated at Hotspots                                     122
		1.15 Many Hotspots Result from Modified Bases                                   123
		1.16 Some Hereditary Agents Are Extremely Small                                 128
		1.17 Most Genes Encode Polypeptides                                             131
		1.18 Mutations in the Same Gene Cannot Complement                               134
		1.19 Mutations May Cause Loss of Function or Gain of Function                   138
		1.20 A Locus Can Have Many Different Mutant Alleles                             141
		1.21 A Locus Can Have More Than One Wild-Type Allele                            144
		1.22 Recombination Occurs by Physical Exchange of DNA                           146
		1.23 The Genetic Code Is Triplet                                                153
		1.24 Every Coding Sequence Has Three Possible Reading Frames                    159
		1.25 Bacterial Genes Are Colinear with Their Products                           161
		1.26 Several Processes Are Required to Express the Product of a Gene            164
		1.27 Proteins Are trans-Acting but Sites on DNA Are cis-Acting                  169
	Chapter 2 Methods in Molecular Biology and Genetic Engineering                  185
		2.1 Introduction                                                                188
		2.2 Nucleases                                                                   189
		2.3 Cloning                                                                     196
		2.4 Cloning Vectors Can Be Specialized for Different Purposes                   203
		2.5 Nucleic Acid Detection                                                      212
		2.6 DNA Separation Techniques                                                   218
		2.7 DNA Sequencing                                                              227
		2.8 PCR and RT-PCR                                                              231
		2.9 Blotting Methods                                                            245
		2.10 DNA Microarrays                                                            255
		2.11 Chromatin Immunoprecipitation                                              261
		2.12 Gene Knockouts, Transgenics, and Genome Editing                            265
	Chapter 3 The Interrupted Gene                                                  289
		3.1 Introduction                                                                292
		3.2 An Interrupted Gene Has Exons and Introns                                   295
		3.3 Exon and Intron Base Compositions Differ                                    297
		3.4 Organization of Interrupted Genes Can Be Conserved                          299
		3.5 Exon Sequences Under Negative Selection Are Conserved but Introns Vary      303
		3.6 Exon Sequences Under Positive Selection Vary but Introns Are Conserved      307
		3.7 Genes Show a Wide Distribution of Sizes Due Primarily to Intron Size and Number Variation    309
		3.8 Some DNA Sequences Encode More Than One Polypeptide                         314
		3.9 Some Exons Correspond to Protein Functional Domains                         320
		3.10 Members of a Gene Family Have a Common Organization                        324
		3.11 There Are Many Forms of Information in DNA                                 330
	Chapter 4 The Content of the Genome                                             347
		4.1 Introduction                                                                350
		4.2 Genome Mapping Reveals That Individual Genomes Show Extensive Variation     353
		4.3 SNPs Can Be Associated with Genetic Disorders                               357
		4.4 Eukaryotic Genomes Contain Nonrepetitive and Repetitive DNA Sequences       361
		4.5 Eukaryotic Protein-Coding Genes Can Be Identified by the Conservation of Exons and of Genome Organization    365
		4.6 Some Eukaryotic Organelles Have DNA                                         372
		4.7 Organelle Genomes Are Circular DNAs That Encode Organelle Proteins          378
		4.8 The Chloroplast Genome Encodes Many Proteins and RNAs                       386
		4.9 Mitochondria and Chloroplasts Evolved by Endosymbiosis                      389
	Chapter 5 Genome Sequences and Evolution                                        403
		5.1 Introduction                                                                407
		5.2 Prokaryotic Gene Numbers Range Over an Order of Magnitude                   412
		5.3 Total Gene Number Is Known for Several Eukaryotes                           416
		5.4 How Many Different Types of Genes Are There?                                422
		5.5 The Human Genome Has Fewer Genes Than Originally Expected                   429
		5.6 How Are Genes and Other Sequences Distributed in the Genome?                434
		5.7 The Y Chromosome Has Several Male-Specific Genes                            438
		5.8 How Many Genes Are Essential?                                               441
		5.9 About 10,000 Genes Are Expressed at Widely Differing Levels in a Eukaryotic Cell    450
		5.10 Expressed Gene Number Can Be Measured En Masse                             455
		5.11 DNA Sequences Evolve by Mutation and a Sorting Mechanism                   459
		5.12 Selection Can Be Detected by Measuring Sequence Variation                  465
		5.13 A Constant Rate of Sequence Divergence Is a Molecular Clock                475
		5.14 The Rate of Neutral Substitution Can Be Measured from Divergence of Repeated Sequences    485
		5.15 How Did Interrupted Genes Evolve?                                          488
		5.16 Why Are Some Genomes So Large?                                             496
		5.17 Morphological Complexity Evolves by Adding New Gene Functions              502
		5.18 Gene Duplication Contributes to Genome Evolution                           506
		5.19 Globin Clusters Arise by Duplication and Divergence                        509
		5.20 Pseudogenes Have Lost Their Original Functions                             515
		5.21 Genome Duplication Has Played a Role in Plant and Vertebrate Evolution     520
		5.22 What Is the Role of Transposable Elements in Genome Evolution              524
		5.23 There Can Be Biases in Mutation, Gene Conversion, and Codon Usage          526
	Chapter 6 Clusters and Repeats                                                  555
		6.1 Introduction                                                                558
		6.2 Unequal Crossing-Over Rearranges Gene Clusters                              564
		6.3 Genes for rRNA Form Tandem Repeats Including an Invariant Transcription Unit    572
		6.4 Crossover Fixation Could Maintain Identical Repeats                         580
		6.5 Satellite DNAs Often Lie in Heterochromatin                                 588
		6.6 Arthropod Satellites Have Very Short Identical Repeats                      593
		6.7 Mammalian Satellites Consist of Hierarchical Repeats                        595
		6.8 Minisatellites Are Useful for DNA Profiling                                 606
	Chapter 7 Chromosomes                                                           617
		7.1 Introduction                                                                621
		7.2 Viral Genomes Are Packaged into Their Coats                                 624
		7.3 The Bacterial Genome Is a Nucleoid with Dynamic Structural Properties       633
		7.4 The Bacterial Genome Is Supercoiled and Has Four Macrodomains               640
		7.5 Eukaryotic DNA Has Loops and Domains Attached to a Scaffold                 644
		7.6 Specific Sequences Attach DNA to an Interphase Matrix                       647
		7.7 Chromatin Is Divided into Euchromatin and Heterochromatin                   650
		7.8 Chromosomes Have Banding Patterns                                           658
		7.9 Lampbrush Chromosomes Are Extended                                          662
		7.10 Polytene Chromosomes Form Bands                                            665
		7.11 Polytene Chromosomes Expand at Sites of Gene Expression                    670
		7.12 The Eukaryotic Chromosome Is a Segregation Device                          674
		7.13 Regional Centromeres Contain a Centromeric Histone H3 Variant and Repetitive DNA    677
		7.14 Point Centromeres in S. cerevisiae Contain Short, Essential DNA Sequences    682
		7.15 The S. cerevisiae Centromere Binds a Protein Complex                       684
		7.16 Telomeres Have Simple Repeating Sequences                                  687
		7.17 Telomeres Seal the Chromosome Ends and Function in Meiotic Chromosome Pairing    690
		7.18 Telomeres Are Synthesized by a Ribonucleoprotein Enzyme                    696
		7.19 Telomeres Are Essential for Survival                                       701
	Chapter 8 Chromatin                                                             727
		8.1 Introduction                                                                730
		8.2 DNA Is Organized in Arrays of Nucleosomes                                   733
		8.3 The Nucleosome Is the Subunit of All Chromatin                              740
		8.4 Nucleosomes Are Covalently Modified                                         753
		8.5 Histone Variants Produce Alternative Nucleosomes                            767
		8.6 DNA Structure Varies on the Nucleosomal Surface                             772
		8.7 The Path of Nucleosomes in the Chromatin Fiber                              781
		8.8 Replication of Chromatin Requires Assembly of Nucleosomes                   787
		8.9 Do Nucleosomes Lie at Specific Positions?                                   794
		8.10 Nucleosomes Are Displaced and Reassembled During Transcription             807
		8.11 DNase Sensitivity Detects Changes in Chromatin Structure                   817
		8.12 An LCR Can Control a Domain                                                823
		8.13 Insulators Define Transcriptionally Independent Domains                    829
PART II DNA Replication and Recombination                                       866
	Chapter 9 Replication Is Connected to the Cell Cycle                            869
		9.1 Introduction                                                                872
		9.2 Bacterial Replication Is Connected to the Cell Cycle                        877
		9.3 The Shape and Spatial Organization of a Bacterium Are Important During Chromosome Segregation and Cell Division    880
		9.4 Mutations in Division or Segregation Affect Cell Shape                      883
		9.5 FtsZ Is Necessary for Septum Formation                                      886
		9.6 min and noc/slm Genes Regulate the Location of the Septum                   888
		9.7 Partition Involves Separation of the Chromosomes                            892
		9.8 Chromosomal Segregation Might Require Site-Specific Recombination           895
		9.9 The Eukaryotic Growth Factor Signal Transduction Pathway Promotes Entry to S Phase    901
		9.10 Checkpoint Control for Entry into S Phase: p53, a Guardian of the Checkpoint    908
		9.11 Checkpoint Control for Entry into S Phase: Rb, a Guardian of the Checkpoint    913
	Chapter 10 The Replicon: Initiation of Replication                              933
		10.1 Introduction                                                               936
		10.2 An Origin Usually Initiates Bidirectional Replication                      939
		10.3 The Bacterial Genome Is (Usually) a Single Circular Replicon               943
		10.4 Methylation of the Bacterial Origin Regulates Initiation                   945
		10.5 Initiation: Creating the Replication Forks at the Origin oriC              949
		10.6 Multiple Mechanisms Exist to Prevent Premature Reinitiation of Replication    955
		10.7 Archaeal Chromosomes Can Contain Multiple Replicons                        959
		10.8 Each Eukaryotic Chromosome Contains Many Replicons                         959
		10.9 Replication Origins Can Be Isolated in Yeast                               964
		10.10 Licensing Factor Controls Eukaryotic Rereplication                        968
		10.11 Licensing Factor Binds to ORC                                             972
	Chapter 11 DNA Replication                                                      991
		11.1 Introduction                                                               994
		11.2 DNA Polymerases Are the Enzymes That Make DNA                              997
		11.3 DNA Polymerases Have Various Nuclease Activities                           1002
		11.4 DNA Polymerases Control the Fidelity of Replication                        1004
		11.5 DNA Polymerases Have a Common Structure                                    1010
		11.6 The Two New DNA Strands Have Different Modes of Synthesis                  1013
		11.7 Replication Requires a Helicase and a Single-Stranded Binding Protein      1016
		11.8 Priming Is Required to Start DNA Synthesis                                 1019
		11.9 Coordinating Synthesis of the Lagging and Leading Strands                  1025
		11.10 DNA Polymerase Holoenzyme Consists of Subcomplexes                        1027
		11.11 The Clamp Controls Association of Core Enzyme with DNA                    1031
		11.12 Okazaki Fragments Are Linked by Ligase                                    1040
		11.13 Separate Eukaryotic DNA Polymerases Undertake Initiation and Elongation    1046
		11.14 Lesion Bypass Requires Polymerase Replacement                             1052
		11.15 Termination of Replication                                                1060
	Chapter 12 Extrachromosomal Replicons                                           1075
		12.1 Introduction                                                               1078
		12.2 The Ends of Linear DNA Are a Problem for Replication                       1081
		12.3 Terminal Proteins Enable Initiation at the Ends of Viral DNAs              1083
		12.4 Rolling Circles Produce Multimers of a Replicon                            1088
		12.5 Rolling Circles Are Used to Replicate Phage Genomes                        1093
		12.6 The F Plasmid Is Transferred by Conjugation Between Bacteria               1097
		12.7 Conjugation Transfers Single-Stranded DNA                                  1102
		12.8 Single-Copy Plasmids Have a Partitioning System                            1107
		12.9 Plasmid Incompatibility Is Determined by the Replicon                      1116
		12.10 The ColE1 Compatibility System Is Controlled by an RNA Regulator          1118
		12.11 How Do Mitochondria Replicate and Segregate?                              1125
		12.12 D Loops Maintain Mitochondrial Origins                                    1130
		12.13 The Bacterial Ti Plasmid Causes Crown Gall Disease in Plants              1133
		12.14 T-DNA Carries Genes Required for Infection                                1137
		12.15 Transfer of T-DNA Resembles Bacterial Conjugation                         1145
	Chapter 13 Homologous and Site-Specific Recombination                           1161
		13.1 Introduction                                                               1165
		13.2 Homologous Recombination Occurs Between Synapsed Chromosomes in Meiosis    1169
		13.3 Double-Strand Breaks Initiate Recombination                                1174
		13.4 Gene Conversion Accounts for Interallelic Recombination                    1183
		13.5 The Synthesis-Dependent Strand-Annealing Model                             1187
		13.6 The Single-Strand Annealing Mechanism Functions at Some Double-Strand Breaks    1190
		13.7 Break-Induced Replication Can Repair Double-Strand Breaks                  1193
		13.8 Recombining Meiotic Chromosomes Are Connected by the Synaptonemal Complex    1196
		13.9 The Synaptonemal Complex Forms After Double-Strand Breaks                  1200
		13.10 Pairing and Synaptonemal Complex Formation Are Independent                1206
		13.11 The Bacterial RecBCD System Is Stimulated by chi Sequences                1208
		13.12 Strand-Transfer Proteins Catalyze Single-Strand Assimilation              1212
		13.13 Holliday Junctions Must Be Resolved                                       1221
		13.14 Eukaryotic Genes Involved in Homologous Recombination                     1226
			1. End Processing/Presynapsis                                                   1227
			2. Synapsis                                                                     1233
			3. DNA Heteroduplex Extension and Branch Migration                              1234
			4. Resolution                                                                   1234
		13.15 Specialized Recombination Involves Specific Sites                         1236
		13.16 Site-Specific Recombination Involves Breakage and Reunion                 1241
		13.17 Site-Specific Recombination Resembles Topoisomerase Activity              1243
		13.18 Lambda Recombination Occurs in an Intasome                                1248
		13.19 Yeast Can Switch Silent and Active Mating-Type Loci                       1252
		13.20 Unidirectional Gene Conversion Is Initiated by the Recipient MAT Locus    1258
		13.21 Antigenic Variation in Trypanosomes Uses Homologous Recombination         1262
		13.22 Recombination Pathways Adapted for Experimental Systems                   1265
	Chapter 14 Repair Systems                                                       1292
		14.1 Introduction                                                               1295
		14.2 Repair Systems Correct Damage to DNA                                       1300
		14.3 Excision Repair Systems in E. coli                                         1307
		14.4 Eukaryotic Nucleotide Excision Repair Pathways                             1312
		14.5 Base Excision Repair Systems Require Glycosylases                          1317
		14.6 Error-Prone Repair and Translesion Synthesis                               1325
		14.7 Controlling the Direction of Mismatch Repair                               1327
		14.8 Recombination-Repair Systems in E. coli                                    1336
		14.9 Recombination Is an Important Mechanism to Recover from Replication Errors    1340
		14.10 Recombination-Repair of Double-Strand Breaks in Eukaryotes                1346
		14.11 Nonhomologous End Joining Also Repairs Double-Strand Breaks               1349
		14.12 DNA Repair in Eukaryotes Occurs in the Context of Chromatin               1355
		14.13 RecA Triggers the SOS System                                              1363
	Chapter 15 Transposable Elements and Retroviruses                               1391
		15.1 Introduction                                                               1395
		15.2 Insertion Sequences Are Simple Transposition Modules                       1400
		15.3 Transposition Occurs by Both Replicative and Nonreplicative Mechanisms     1404
		15.4 Transposons Cause Rearrangement of DNA                                     1408
		15.5 Replicative Transposition Proceeds Through a Cointegrate                   1412
		15.6 Nonreplicative Transposition Proceeds by Breakage and Reunion              1417
		15.7 Transposons Form Superfamilies and Families                                1422
		15.8 The Role of Transposable Elements in Hybrid Dysgenesis                     1432
		15.9 P Elements Are Activated in the Germline                                   1435
		15.10 The Retrovirus Life Cycle Involves Transposition-Like Events              1440
		15.11 Retroviral Genes Code for Polyproteins                                    1443
		15.12 Viral DNA Is Generated by Reverse Transcription                           1446
		15.13 Viral DNA Integrates into the Chromosome                                  1455
		15.14 Retroviruses May Transduce Cellular Sequences                             1460
		15.15 Retroelements Fall into Three Classes                                     1463
		15.16 Yeast Ty Elements Resemble Retroviruses                                   1469
		15.17 The Alu Family Has Many Widely Dispersed Members                          1475
		15.18 LINEs Use an Endonuclease to Generate a Priming End                       1477
	Chapter 16 Somatic DNA Recombination and Hypermutation in the Immune System     1503
		16.1 The Immune System: Innate and Adaptive Immunity                            1507
		16.2 The Innate Response Utilizes Conserved Recognition Molecules and Signaling Pathways    1511
		16.3 Adaptive Immunity                                                          1517
		16.4 Clonal Selection Amplifies Lymphocytes That Respond to a Given Antigen     1522
		16.5 Ig Genes Are Assembled from Discrete DNA Segments in B Lymphocytes         1527
		16.6 L Chains Are Assembled by a Single Recombination Event                     1531
		16.7 H Chains Are Assembled by Two Sequential Recombination Events              1534
		16.8 Recombination Generates Extensive Diversity                                1537
		16.9 V(D)J DNA Recombination Relies on RSS and Occurs by Deletion or Inversion    1540
		16.10 Allelic Exclusion Is Triggered by Productive Rearrangements               1545
		16.11 RAG1/RAG2 Catalyze Breakage and Religation of V(D)J Gene Segments         1549
		16.12 B Cell Development in the Bone Marrow: From Common Lymphoid Progenitor to Mature B Cell    1556
		16.13 Class Switch DNA Recombination                                            1562
		16.14 CSR Involves AID and Elements of the NHEJ Pathway                         1568
		16.15 Somatic Hypermutation Generates Additional Diversity and Provides the Substrate for Higher-Affinity Submutants    1573
		16.16 SHM Is Mediated by AID, Ung, Elements of the Mismatch DNA Repair Machinery, and Translesion DNA Synthesis Polymerases    1578
		16.17 Igs Expressed in Avians Are Assembled from Pseudogenes                    1581
		16.18 Chromatin Architecture Dynamics of the IgH Locus in V(D)J Recombination, CSR, and SHM    1584
		16.19 Epigenetics of V(D)J Recombination, CSR, and SHM                          1587
		16.20 B Cell Differentiation Results in Maturation of the Antibody Response and Generation of Long-lived Plasma Cells and Memory B Cells    1593
		16.21 The T Cell Receptor Antigen Is Related to the BCR                         1597
		16.22 The TCR Functions in Conjunction with the MHC                             1602
		16.23 The MHC Locus Comprises a Cohort of Genes Involved in Immune Recognition    1607
PART III Transcription and Posttranscriptional Mechanisms                       1671
	Chapter 17 Prokaryotic Transcription                                            1674
		17.1 Introduction                                                               1678
		17.2 Transcription Occurs by Base Pairing in a "Bubble" of Unpaired DNA         1682
		17.3 The Transcription Reaction Has Three Stages                                1685
		17.4 Bacterial RNA Polymerase Consists of Multiple Subunits                     1689
		17.5 RNA Polymerase Holoenzyme Consists of the Core Enzyme and Sigma Factor     1694
		17.6 How Does RNA Polymerase Find Promoter Sequences?                           1696
		17.7 The Holoenzyme Goes Through Transitions in the Process of Recognizing and Escaping from Promoters    1698
		17.8 Sigma Factor Controls Binding to DNA by Recognizing Specific Sequences in Promoters    1703
		17.9 Promoter Efficiencies Can Be Increased or Decreased by Mutation            1709
		17.10 Multiple Regions in RNA Polymerase Directly Contact Promoter DNA          1712
		17.11 RNA Polymerase—Promoter and DNA—Protein Interactions Are the Same for Promoter Recognition and DNA Melting    1721
		17.12 Interactions Between Sigma Factor and Core RNA Polymerase Change During Promoter Escape    1728
		17.13 A Model for Enzyme Movement Is Suggested by the Crystal Structure         1731
		17.14 A Stalled RNA Polymerase Can Restart                                      1736
		17.15 Bacterial RNA Polymerase Terminates at Discrete Sites                     1738
		17.16 How Does Rho Factor Work?                                                 1744
		17.17 Supercoiling Is an Important Feature of Transcription                     1750
		17.18 Phage T7 RNA Polymerase Is a Useful Model System                          1753
		17.19 Competition for Sigma Factors Can Regulate Initiation                     1755
		17.20 Sigma Factors Can Be Organized into Cascades                              1760
		17.21 Sporulation Is Controlled by Sigma Factors                                1764
		17.22 Antitermination Can Be a Regulatory Event                                 1772
	Chapter 18 Eukaryotic Transcription                                             1807
		18.1 Introduction                                                               1810
		18.2 Eukaryotic RNA Polymerases Consist of Many Subunits                        1816
		18.3 RNA Polymerase I Has a Bipartite Promoter                                  1820
		18.4 RNA Polymerase III Uses Downstream and Upstream Promoters                  1824
		18.5 The Start Point for RNA Polymerase II                                      1831
		18.6 TBP Is a Universal Factor                                                  1834
		18.7 The Basal Apparatus Assembles at the Promoter                              1842
		18.8 Initiation Is Followed by Promoter Clearance and Elongation                1850
		18.9 Enhancers Contain Bidirectional Elements That Assist Initiation            1858
		18.10 Enhancers Work by Increasing the Concentration of Activators Near the Promoter    1862
		18.11 Gene Expression Is Associated with Demethylation                          1865
		18.12 CpG Islands Are Regulatory Targets                                        1869
	Chapter 19 RNA Splicing and Processing                                          1900
		19.1 Introduction                                                               1903
		19.2 The 5ʹ End of Eukaryotic mRNA Is Capped                                    1907
		19.3 Nuclear Splice Sites Are Short Sequences                                   1910
		19.4 Splice Sites Are Read in Pairs                                             1913
		19.5 Pre-mRNA Splicing Proceeds Through a Lariat                                1917
		19.6 snRNAs Are Required for Splicing                                           1921
		19.7 Commitment of Pre-mRNA to the Splicing Pathway                             1925
		19.8 The Spliceosome Assembly Pathway                                           1932
		19.9 An Alternative Spliceosome Uses Different snRNPs to Process the Minor Class of Introns    1940
		19.10 Pre-mRNA Splicing Likely Shares the Mechanism with Group II Autocatalytic Introns    1943
		19.11 Splicing Is Temporally and Functionally Coupled with Multiple Steps in Gene Expression    1948
		19.12 Alternative Splicing Is a Rule, Rather Than an Exception, in Multicellular Eukaryotes    1957
		19.13 Splicing Can Be Regulated by Exonic and Intronic Splicing Enhancers and Silencers    1962
		19.14 trans-Splicing Reactions Use Small RNAs                                   1968
		19.15 The 3ʹ Ends of mRNAs Are Generated by Cleavage and Polyadenylation        1974
		19.16 3ʹ mRNA End Processing Is Critical for Termination of Transcription       1979
		19.17 The 3ʹ End Formation of Histone mRNA Requires U7 snRNA                    1983
		19.18 tRNA Splicing Involves Cutting and Rejoining in Separate Reactions        1987
		19.19 The Unfolded Protein Response Is Related to tRNA Splicing                 1995
		19.20 Production of rRNA Requires Cleavage Events and Involves Small RNAs       1998
	Chapter 20 mRNA Stability and Localization                                      2038
		20.1 Introduction                                                               2041
		20.2 Messenger RNAs Are Unstable Molecules                                      2044
		20.3 Eukaryotic mRNAs Exist in the Form of mRNPs from Their Birth to Their Death    2049
		20.4 Prokaryotic mRNA Degradation Involves Multiple Enzymes                     2052
		20.5 Most Eukaryotic mRNA Is Degraded via Two Deadenylation-Dependent Pathways    2057
		20.6 Other Degradation Pathways Target Specific mRNAs                           2062
		20.7 mRNA-Specific Half-Lives Are Controlled by Sequences or Structures Within the mRNA    2067
		20.8 Newly Synthesized RNAs Are Checked for Defects via a Nuclear Surveillance System    2072
		20.9 Quality Control of mRNA Translation Is Performed by Cytoplasmic Surveillance Systems    2076
		20.10 Translationally Silenced mRNAs Are Sequestered in a Variety of RNA Granules    2083
		20.11 Some Eukaryotic mRNAs Are Localized to Specific Regions of a Cell         2086
	Chapter 21 Catalytic RNA                                                        2110
		21.1 Introduction                                                               2113
		21.2 Group I Introns Undertake Self-Splicing by Transesterification             2116
		21.3 Group I Introns Form a Characteristic Secondary Structure                  2124
		21.4 Ribozymes Have Various Catalytic Activities                                2129
		21.5 Some Group I Introns Encode Endonucleases That Sponsor Mobility            2137
		21.6 Group II Introns May Encode Multifunction Proteins                         2141
		21.7 Some Autosplicing Introns Require Maturases                                2143
		21.8 The Catalytic Activity of RNase P Is Due to RNA                            2146
		21.9 Viroids Have Catalytic Activity                                            2148
		21.10 RNA Editing Occurs at Individual Bases                                    2153
		21.11 RNA Editing Can Be Directed by Guide RNAs                                 2157
		21.12 Protein Splicing Is Autocatalytic                                         2164
	Chapter 22 Translation                                                          2187
		22.1 Introduction                                                               2191
		22.2 Translation Occurs by Initiation, Elongation, and Termination              2194
		22.3 Special Mechanisms Control the Accuracy of Translation                     2201
		22.4 Initiation in Bacteria Needs 30S Subunits and Accessory Factors            2204
		22.5 Initiation Involves Base Pairing Between mRNA and rRNA                     2210
		22.6 A Special Initiator tRNA Starts the Polypeptide Chain                      2214
		22.7 Use of fMet-tRNAf Is Controlled by IF-2 and the Ribosome                   2219
		22.8 Small Subunits Scan for Initiation Sites on Eukaryotic mRNA                2222
		22.9 Eukaryotes Use a Complex of Many Initiation Factors                        2227
		22.10 Elongation Factor Tu Loads Aminoacyl-tRNA into the A Site                 2236
		22.11 The Polypeptide Chain Is Transferred to Aminoacyl-tRNA                    2241
		22.12 Translocation Moves the Ribosome                                          2244
		22.13 Elongation Factors Bind Alternately to the Ribosome                       2248
		22.14 Three Codons Terminate Translation                                        2253
		22.15 Termination Codons Are Recognized by Protein Factors                      2255
		22.16 Ribosomal RNA Is Found Throughout Both Ribosomal Subunits                 2264
		22.17 Ribosomes Have Several Active Centers                                     2271
		22.18 16S rRNA Plays an Active Role in Translation                              2279
		22.19 23S rRNA Has Peptidyl Transferase Activity                                2286
		22.20 Ribosomal Structures Change When the Subunits Come Together               2290
		22.21 Translation Can Be Regulated                                              2291
		22.22 The Cycle of Bacterial Messenger RNA                                      2296
	Chapter 23 Using the Genetic Code                                               2329
		23.1 Introduction                                                               2333
		23.2 Related Codons Represent Chemically Similar Amino Acids                    2334
		23.3 Codon—Anticodon Recognition Involves Wobbling                             2338
		23.4 tRNAs Are Processed from Longer Precursors                                 2344
		23.5 tRNA Contains Modified Bases                                               2347
		23.6 Modified Bases Affect Anticodon—Codon Pairing                             2352
		23.7 The Universal Code Has Experienced Sporadic Alterations                    2355
		23.8 Novel Amino Acids Can Be Inserted at Certain Stop Codons                   2360
		23.9 tRNAs Are Charged with Amino Acids by Aminoacyl-tRNA Synthetases           2364
		23.10 Aminoacyl-tRNA Synthetases Fall into Two Classes                          2370
		23.11 Synthetases Use Proofreading to Improve Accuracy                          2376
		23.12 Suppressor tRNAs Have Mutated Anticodons That Read New Codons             2383
		23.13 Each Termination Codon Has Nonsense Suppressors                           2387
		23.14 Suppressors May Compete with Wild-Type Reading of the Code                2390
		23.15 The Ribosome Influences the Accuracy of Translation                       2394
		23.16 Frameshifting Occurs at Slippery Sequences                                2400
		23.17 Other Recoding Events: Translational Bypassing and the tmRNA Mechanism to Free Stalled Ribosomes    2406
PART IV Gene Regulation                                                         2426
	Chapter 24 The Operon                                                           2429
		24.1 Introduction                                                               2433
		24.2 Structural Gene Clusters Are Coordinately Controlled                       2441
		24.3 The lac Operon Is Negative Inducible                                       2443
		24.4 The lac Repressor Is Controlled by a Small-Molecule Inducer                2448
		24.5 cis-Acting Constitutive Mutations Identify the Operator                    2452
		24.6 trans-Acting Mutations Identify the Regulator Gene                         2455
		24.7 The lac Repressor Is a Tetramer Made of Two Dimers                         2459
		24.8 lac Repressor Binding to the Operator Is Regulated by an Allosteric Change in Conformation    2467
		24.9 The lac Repressor Binds to Three Operators and Interacts with RNA Polymerase    2472
		24.10 The Operator Competes with Low-Affinity Sites to Bind Repressor           2477
		24.11 The lac Operon Has a Second Layer of Control: Catabolite Repression       2481
		24.12 The trp Operon Is a Repressible Operon with Three Transcription Units     2490
		24.13 The trp Operon Is Also Controlled by Attenuation                          2494
		24.14 Attenuation Can Be Controlled by Translation                              2498
		24.15 Stringent Control by Stable RNA Transcription                             2505
		24.16 r-Protein Synthesis Is Controlled by Autoregulation                       2509
	Chapter 25 Phage Strategies                                                     2531
		25.1 Introduction                                                               2534
		25.2 Lytic Development Is Divided into Two Periods                              2538
		25.3 Lytic Development Is Controlled by a Cascade                               2541
		25.4 Two Types of Regulatory Events Control the Lytic Cascade                   2547
		25.5 The Phage T7 and T4 Genomes Show Functional Clustering                     2549
		25.6 Lambda Immediate Early and Delayed Early Genes Are Needed for Both Lysogeny and the Lytic Cycle    2553
		25.7 The Lytic Cycle Depends on Antitermination by pN                           2556
		25.8 Lysogeny Is Maintained by the Lambda Repressor Protein                     2561
		25.9 The Lambda Repressor and Its Operators Define the Immunity Region          2564
		25.10 The DNA-Binding Form of the Lambda Repressor Is a Dimer                   2566
		25.11 The Lambda Repressor Uses a Helix-Turn-Helix Motif to Bind DNA            2570
		25.12 Lambda Repressor Dimers Bind Cooperatively to the Operator                2575
		25.13 The Lambda Repressor Maintains an Autoregulatory Circuit                  2579
		25.14 Cooperative Interactions Increase the Sensitivity of Regulation           2583
		25.15 The cII and cIII Genes Are Needed to Establish Lysogeny                   2585
		25.16 A Poor Promoter Requires cII Protein                                      2588
		25.17 Lysogeny Requires Several Events                                          2591
		25.18 The Cro Repressor Is Needed for Lytic Infection                           2594
		25.19 What Determines the Balance Between Lysogeny and the Lytic Cycle?         2598
	Chapter 26 Eukaryotic Transcription Regulation                                  2614
		26.1 Introduction                                                               2618
		26.2 How Is a Gene Turned On?                                                   2622
		26.3 Mechanism of Action of Activators and Repressors                           2625
		26.4 Independent Domains Bind DNA and Activate Transcription                    2633
		26.5 The Two-Hybrid Assay Detects Protein—Protein Interactions                 2635
		26.6 Activators Interact with the Basal Apparatus                               2637
		26.7 Many Types of DNA-Binding Domains Have Been Identified                     2645
		26.8 Chromatin Remodeling Is an Active Process                                  2651
		26.9 Nucleosome Organization or Content Can Be Changed at the Promoter          2658
		26.10 Histone Acetylation Is Associated with Transcription Activation           2665
		26.11 Methylation of Histones and DNA Is Connected                              2674
		26.12 Promoter Activation Involves Multiple Changes to Chromatin                2678
		26.13 Histone Phosphorylation Affects Chromatin Structure                       2682
		26.14 Yeast GAL Genes: A Model for Activation and Repression                    2685
	Chapter 27 Epigenetics I                                                        2721
		27.1 Introduction                                                               2724
		27.2 Heterochromatin Propagates from a Nucleation Event                         2728
		27.3 Heterochromatin Depends on Interactions with Histones                      2733
		27.4 Polycomb and Trithorax Are Antagonistic Repressors and Activators          2743
		27.5 CpG Islands Are Subject to Methylation                                     2749
		27.6 Epigenetic Effects Can Be Inherited                                        2759
		27.7 Yeast Prions Show Unusual Inheritance                                      2766
	Chapter 28 Epigenetics II                                                       2798
		28.1 Introduction                                                               2802
		28.2 X Chromosomes Undergo Global Changes                                       2803
		28.3 Chromosome Condensation Is Caused by Condensins                            2813
		28.4 DNA Methylation Is Responsible for Imprinting                              2823
		28.5 Oppositely Imprinted Genes Can Be Controlled by a Single Center            2829
		28.6 Prions Cause Diseases in Mammals                                           2831
	Chapter 29 Noncoding RNA                                                        2851
		29.1 Introduction                                                               2854
		29.2 A Riboswitch Can Alter Its Structure According to Its Environment          2855
		29.3 Noncoding RNAs Can Be Used to Regulate Gene Expression                     2860
	Chapter 30 Regulatory RNA                                                       2878
		30.1 Introduction                                                               2880
		30.2 Bacteria Contain Regulator RNAs                                            2882
		30.3 MicroRNAs Are Widespread Regulators in Eukaryotes                          2889
		30.4 How Does RNA Interference Work?                                            2898
		30.5 Heterochromatin Formation Requires MicroRNAs                               2906
Glossary                                                                        2925
Index                                                                           3050
```

